[2025-03-13T04:01:18.626+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-13T04:01:18.736+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: data_ingestion_gcs_dag.format_parquet_2024_q3 manual__2025-03-13T04:00:50.830640+00:00 [queued]>
[2025-03-13T04:01:18.778+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: data_ingestion_gcs_dag.format_parquet_2024_q3 manual__2025-03-13T04:00:50.830640+00:00 [queued]>
[2025-03-13T04:01:18.787+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 2
[2025-03-13T04:01:18.833+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): format_parquet_2024_q3> on 2025-03-13 04:00:50.830640+00:00
[2025-03-13T04:01:18.850+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=1007) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-03-13T04:01:18.858+0000] {standard_task_runner.py:72} INFO - Started process 1028 to run task
[2025-03-13T04:01:18.864+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'data_ingestion_gcs_dag', 'format_parquet_2024_q3', 'manual__2025-03-13T04:00:50.830640+00:00', '--job-id', '488', '--raw', '--subdir', 'DAGS_FOLDER/data_ingestion_gcs_dag.py', '--cfg-path', '/tmp/tmpgcfoi7yw']
[2025-03-13T04:01:18.870+0000] {standard_task_runner.py:105} INFO - Job 488: Subtask format_parquet_2024_q3
[2025-03-13T04:01:19.058+0000] {task_command.py:467} INFO - Running <TaskInstance: data_ingestion_gcs_dag.format_parquet_2024_q3 manual__2025-03-13T04:00:50.830640+00:00 [running]> on host b0577ed9eb82
[2025-03-13T04:01:19.619+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='data_ingestion_gcs_dag' AIRFLOW_CTX_TASK_ID='format_parquet_2024_q3' AIRFLOW_CTX_EXECUTION_DATE='2025-03-13T04:00:50.830640+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-13T04:00:50.830640+00:00'
[2025-03-13T04:01:19.627+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-03-13T04:01:19.628+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-03-13T04:01:19.633+0000] {logging_mixin.py:190} INFO - Current task name:format_parquet_2024_q3 state:running start_date:2025-03-13 04:01:18.742212+00:00
[2025-03-13T04:01:19.636+0000] {logging_mixin.py:190} INFO - Dag name:data_ingestion_gcs_dag and current dag run status:running
[2025-03-13T04:01:19.639+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-13T04:01:23.154+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/data_ingestion_gcs_dag.py", line 53, in format_to_parquet
    table = table.append_column("year", pv.array([str(year)] * table.num_rows))
                                        ^^^^^^^^
AttributeError: module 'pyarrow.csv' has no attribute 'array'
[2025-03-13T04:01:23.392+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-03-13T04:01:23.401+0000] {logging_mixin.py:190} INFO - Task start:2025-03-13 04:01:18.742212+00:00 end:2025-03-13 04:01:23.387185+00:00 duration:4.644973
[2025-03-13T04:01:23.409+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): format_parquet_2024_q3> dag:<DAG: data_ingestion_gcs_dag> dagrun:<DagRun data_ingestion_gcs_dag @ 2025-03-13 04:00:50.830640+00:00: manual__2025-03-13T04:00:50.830640+00:00, state:running, queued_at: 2025-03-13 04:00:50.872907+00:00. externally triggered: True>
[2025-03-13T04:01:23.410+0000] {logging_mixin.py:190} INFO - Failure caused by module 'pyarrow.csv' has no attribute 'array'
[2025-03-13T04:01:23.416+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=data_ingestion_gcs_dag, task_id=format_parquet_2024_q3, run_id=manual__2025-03-13T04:00:50.830640+00:00, execution_date=20250313T040050, start_date=20250313T040118, end_date=20250313T040123
[2025-03-13T04:01:23.601+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-13T04:01:23.611+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 488 for task format_parquet_2024_q3 (module 'pyarrow.csv' has no attribute 'array'; 1028)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/data_ingestion_gcs_dag.py", line 53, in format_to_parquet
    table = table.append_column("year", pv.array([str(year)] * table.num_rows))
                                        ^^^^^^^^
AttributeError: module 'pyarrow.csv' has no attribute 'array'
[2025-03-13T04:01:23.741+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-13T04:01:23.874+0000] {standard_task_runner.py:217} INFO - Process not found (most likely exited), stop collecting metrics
[2025-03-13T04:01:23.884+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-13T04:01:23.890+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
